LavalinkServer/src/main/java/lavalink/server/bootstrap/PluginManager.kt

```diff
@@ -47,7 +47,10 @@ class PluginManager(val config: PluginsConfig) {
         data class Declaration(val group: String, val name: String, val version: String, val repository: String)
 
         val declarations = config.plugins.map { declaration ->
-            if (declaration.dependency == null || declaration.repository == null) throw RuntimeException("Illegal declaration $declaration")
+            if (declaration.dependency == null) throw RuntimeException("Illegal dependency declaration: null")
+            if (declaration.repository == null) {
+                declaration.repository = if (declaration.snapshot) config.defaultPluginSnapshotRepository else config.defaultPluginRepository
+            }
             val fragments = declaration.dependency!!.split(":")
             if (fragments.size != 3) throw RuntimeException("Invalid dependency \"${declaration.dependency}\"")
             val repository =
```

Comment:
I think it's generally bad practice to treat the configuration properties as mutable objects to be changed. That's why the code has always used to `repository` val below
```suggestion
            if (declaration.dependency == null) throw RuntimeException("Illegal dependency declaration: null")
            val fragments = declaration.dependency!!.split(":")
            if (fragments.size != 3) throw RuntimeException("Invalid dependency \"${declaration.dependency}\"")

            var repository = declaration.repository
                ?: if (declaration.snapshot) config.defaultPluginSnapshotRepository else config.defaultPluginRepository
            repository =
```